### DISCUSSION TITLE
Exploring Languages for Developing a Parallel Integer Linear Programming Solver: Gleam as a Potential Choice

### DISCUSSION THREAD

**Discussion by codeYann** [gleam-lang/gleam#3538] [Ideas & suggestions]
I’m currently exploring languages for developing an Integer Linear Programming (ILP) solver, and I came across Gleam as a potential candidate. My goal is to leverage parallel programming to enhance the solver's performance and scalability.

Objectives:

- Develop an efficient ILP solver, focusing on parallel programming to optimize performance.
- Implement parallelization techniques to distribute computational tasks across multiple cores.
- Ensure that the solver is scalable and effective for complex optimization problems.

I am considering implementing the core solver in some language, while using Python as the interface for easier interaction and integration.

I’m looking to understand if Gleam is the right language for this type of project. Any feedback on its suitability, particularly in terms of handling parallel computing and optimization problems, would be highly valuable.

**Comment by lpil** [comment]
Hello! I don't know what ILP is, but it sounds like it might involve CPU bound numerical computation. The BEAM is optimised for IO and makes some design decisions which makes numerical computation slower, such as using bigints. You may find that another language will result in similar developer experience with superior performance for your problem.

**Reply by codeYann** [reply]
Thank you so much! I'll keep looking for an appropriate language for my project.
